samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -141,6 +141,142 @@ CDecodingPipeline::~CDecodingPipeline()
     Close();
 }
 
+#if (defined(_WIN64) || defined(_WIN32)) && (MFX_VERSION >= MFX_VERSION_NEXT)
+mfxU32 CDecodingPipeline::GetPreferredAdapterNum(const mfxAdaptersInfo & adapters, const sInputParams & params)
```

Comment:
Why this function is in decoder sample? Why not in utilities?

Reply:
Do we actually need WIN protection here and in most other places? API is common between Windows and Linux...

Reply:
since it is implementation, yes we need. I can think about putting it to utility